src/humanloop/eval_utils/run.py

```diff
@@ -0,0 +1,739 @@
+            for page in evals.iter_pages():
+                evaluation = next((e for e in page.items if e.name == name), None)  # type: ignore
+        else:
+            raise error_
+        if not evaluation:
+            raise ValueError(f"Evaluation with name {name} not found.")
+
+    # Create a new Run
+    run: EvaluationRunResponse = client.evaluations.create_run(
+        id=evaluation.id,
+        dataset={"version_id": hl_dataset.version_id},
+        orchestrated=False,
+    )
+    # Every Run will generate a new batch of Logs
+    run_id = run.id
+
+    _PROGRESS_BAR = _SimpleProgressBar(len(hl_dataset.datapoints))  # type: ignore
+
+    if is_decorated:
+
+        def process_datapoint(dp: Datapoint, file_id: str, file_path: str, run_id: str):
+            def upload_callback(log: dict):
+                # OTel exporter will call this after the Log is uploaded
+                _add_log_to_evaluation(
+                    client=client,
+                    log=log,
+                    datapoint_target=dp.target,
+                    local_evaluators=local_evaluators,
+                )
+                _PROGRESS_BAR.increment()  # type: ignore
+
+            datapoint_dict = dp.dict()
+            # Set the Evaluation Context for the Exporter
+            # Each thread will have its own context
+            EVALUATION_CONTEXT.set(
+                EvaluationContext(
+                    source_datapoint_id=dp.id,
+                    upload_callback=upload_callback,
+                    file_id=file_id,
+                    run_id=run_id,
+                    path=file_path,
+                )
+            )
+            if datapoint_dict.get("messages"):
+                # function_ is decorated by Humanloop, the OTel Exporter will
+                # handle the logging, which will call the upload_callback
+                # function above when it's done
+                function_(  # type: ignore
+                    datapoint_dict["inputs"],
+                    messages=datapoint_dict["messages"],
+                )
+            else:
+                # function_ is decorated by Humanloop, the OTel Exporter will
+                # handle the logging, which will call the upload_callback
+                # function above when it's done
+                function_(datapoint_dict["inputs"])  # type: ignore
+
+    else:
+        # Define the function to execute your function in parallel and Log to Humanloop
+        def process_datapoint(dp: Datapoint, file_id: str, file_path: str, run_id: str):
+            log_func = _get_log_func(
+                client=client,
+                file_type=type_,
+                file_id=hl_file.id,
+                version_id=hl_file.version_id,
+                run_id=run_id,
+            )
+
+            start_time = datetime.now()
+            datapoint_dict = dp.dict()
+            try:
+                if "messages" in datapoint_dict:
+                    output = function_(  # type: ignore
```

Comment:
Oh wait, this is still the old behaviour, i'm confused now

Reply:
fixed in this commit
https://github.com/humanloop/humanloop-python/pull/24/commits/c66b3e5321abf40929d3f984df89b91405b7d735